src/tests/Microsoft.Diagnostics.NETCore.Client/TestRunner.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using Xunit;
+using Xunit.Abstractions;
+
+using Microsoft.Diagnostics.TestHelpers;
+
+using Microsoft.Diagnostics.NETCore.Client;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class TestRunner
+    {
+        private Process testProcess;
+        private ProcessStartInfo startInfo;
+        private ITestOutputHelper outputHelper;
+
+        public TestRunner(string testExePath, ITestOutputHelper _outputHelper=null)
+        {
+            startInfo = new ProcessStartInfo(CommonHelper.HostExe, testExePath);
+            startInfo.UseShellExecute = false;
+            startInfo.RedirectStandardOutput = true;
```

Comment:
If the child process logs a bunch of output to stdout and you don't read it then the child will eventually block waiting for it to be read. You'd either need to ensure that no tracee writes too much to stdout, or handle reading it, (or use ProcessRunner which already solved this problem)